spec/ddtrace/benchmark/transport_benchmark_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'spec_helper'
+
+require_relative 'support/benchmark_helper'
+
+require 'socket'
+
+RSpec.describe 'Microbenchmark Transport' do
+  context 'with HTTP transport' do
+    # Create server that responds just like the agent,
+    # but doesn't consume as many resources, nor introduces external
+    # noise into the benchmark.
+    let(:server) { TCPServer.new '127.0.0.1', ENV[Datadog::Ext::Transport::HTTP::ENV_DEFAULT_PORT].to_i }
```

Comment:
How does TCPServer work? Does it run in a separate process?

Reply:
It runs on the same process, but separate thread. Any specific reason for asking?

Reply:
make sense, my reasoning for asking was, introducing something like this in-process, will it impact performance of results measured due to the GIL?

probably not... ? since it all synchronous anyways?

Reply:
I tried with forking. The difference is small, but I think that not boggling the GVL, even if not by very long, is probably the best way to go here.

For JRuby, which doesn't support forking, we still use threads, as it supports true parallel execution in threads.

Reply:
👍🏻